### PR TITLE
Python2.7 support

### PR DESCRIPTION
**Pull request by newey01c**
@cemal01e 
@pope1ni 

Thoughts? :smile:

### CODE REVIEW

File: carmine/mecr.py [hunk 1]
```diff
@@ -128,10 +134,12 @@ def confidence(self):
 
     @property
     def actual_occurrence(self):
-        return len(self.matches)
+        return float(len(self.matches))
 
     @property
     def support(self):
+        print(self.counts)
+        print(self.n_objs)
```
Comment (cemal01e) [inline carmine/mecr.py]: You have a few rouge prints

File: carmine/mecr.py [hunk 2]
```diff
@@ -128,10 +134,12 @@ def confidence(self):
 
     @property
     def actual_occurrence(self):
-        return len(self.matches)
+        return float(len(self.matches))
```
Comment (cemal01e) [inline carmine/mecr.py]: i guess float is no longer necessary because of __future__

File: carmine/mecr.py [hunk 3]
```diff
@@ -17,12 +17,17 @@
...
+    division,
+    print_function,
+    unicode_literals
+)
 
 import numpy as np
 import pandas as pd
 from sklearn.preprocessing import LabelEncoder
-from .rule import Rule, RuleList
+from carmine.rule import Rule, RuleList
```
Comment (cemal01e) [inline carmine/mecr.py]: does making it a relative import break things?

File: carmine/__init__.py [hunk 4]
```diff
@@ -1,4 +1,9 @@
-from __future__ import absolute_import
+from __future__ import (
+    absolute_import,
+    division,
+    print_function,
+    unicode_literals
+)
```
Comment (pope1ni) [inline carmine/__init__.py]: Please put on one line, drop the parentheses.

File: carmine/mecr.py (hunk 3 above)
Comment (pope1ni) [reply carmine/mecr.py]: Please revert these changes.

Relative imports are generally better and no it should not break things. Use of `absolute_import` from `__future__` makes Python 2 behave like Python 3 which fixed some ugliness. If you want to know more, grab me at some point.

File: carmine/mecr.py [hunk 5]
```diff
@@ -66,7 +71,8 @@ def __init__(self, X, y, matches=None):
         self.y = y
 
         # compute number of objects and features in dataset
-        self.n_objs, self.n_feats = X.shape
+        self.n_objs = float(X.shape[0])
+        self.n_feats = X.shape[1]
```
Comment (pope1ni) [inline carmine/mecr.py]: This can be reverted because of the import of `division` from `__future__`.

**Comment by pope1ni** [comment]
Eron, rather. You got predictive texted.

File: carmine/mecr.py (hunk 3 above)
Comment (newey01c) [reply carmine/mecr.py]: Are you sure? PEP8 (& other style guides) seem to prefer absolute imports!

https://www.python.org/dev/peps/pep-0008/#imports

**Pull request merged**